src/main/java/com/assignment1/parser/Parsable.java

```diff
@@ -0,0 +1,15 @@
+package com.assignment1.parser;
+
+import com.assignment1.exceptions.GenericApplicationException;
+import com.assignment1.item.ItemEntity;
+
+/**
+ * Interface for Parser methods.
+ */
+public interface Parsable {
+
+  /**
+   * Method to parse the string.
+   */
+  ItemEntity parse(String... input) throws GenericApplicationException;
```

Comment:
```suggestion
  ItemEntity parse(ItemDTO input) throws GenericApplicationException;
```

Reply:
Made this change with the help of using a converter class